stl/inc/flat_set

```diff
@@ -101,7 +101,7 @@ public:
     template <_Allocator_for<container_type> _Alloc>
     explicit _Base_flat_set(const _Alloc& _Al) : _Mycont(_Al), _Mycomp() {}
 
-    // TRANSITION, an allocator-aware container may not support "C(_First, _Last, _Al)".
+    // FIXME, an allocator-aware container may not support "C(_First, _Last, _Al)".

```

Comment:
From a previous comment (https://github.com/microsoft/STL/pull/4064#discussion_r1342084788):

> I think there is no need to use `make_obj_using_allocator` for `C(another,al) and C(al)`, as these constructors are required for [allocator-aware containers](https://eel.is/c++draft/container.alloc.reqmts). I believe a container that `uses_allocator_v` the allocator should also support these constructors.

These constructors don't seem to be only required with allocator-aware constructors (which is weird).

There may be some non-allocator-aware containers supporting `C(allocator_arg, al, another)`, and we should support them.

> But the `flat_set` constructors only say the container should be constructed with the allocator, then behave the same as without the allocator (inserting the range). So these constructors should be valid even if the container doesn't directly support these constructions.
> 
> > Effects: Equivalent to the corresponding non-allocator constructors except that c is constructed with uses-allocator construction ([[allocator.uses.construction]](https://eel.is/c++draft/allocator.uses.construction))[.](https://eel.is/c++draft/flat.set#cons-10.sentence-1)

I don't see why these constructors should be valid when uses-allocator construction is ill-formed. Note that as of C++20, uses-allocator construction is defined in term of `make_obj_using_allocator` ([[allocator.uses.construction]/1](https://eel.is/c++draft/allocator.uses.construction#1)).

----
I suggest consistently using `make_obj_using_allocator` for `flat_meow` (which is, however, inconsistent with old container adaptors). We may do this in the next PR.

Reply:
>template<class InputIterator>
      flat_set(InputIterator first, InputIterator last, const key_compare& comp = key_compare())
        : c(), compare(comp)
        { insert(first, last); }

Take allocator-version for this constructor for example... Actually most of no-allocator constructors doesn't rely on container's constructors(though they should work fine as they are required by sequential containers)

Reply:
> as only `c(allocator)` is required to be valid

Perhaps this is not even required. Moreover, I don't think flat container adaptors actually require the adapted containers to be allocator-aware anywhere.

> I'm not very sure whether `uses_allocator_v` an allocator-type is able to ensure the validity of `c(allocator)`, though.

The answer is no to me... But given constructors of containers and container adaptors are generally underconstrained, we shouldn't be worried.

Reply:
>c is constructed with uses-allocator construction

Do you think the standard should give more details about this, especially, what should be passed as arguments?

Reply:
> > c is constructed with uses-allocator construction
> 
> Do you think the standard should give more details about this, especially, what should be passed as arguments?

Yes, perhaps there should be an editorial or LWG issue.

I think the current specification should be interpreted as mapping `c(args...)` to `c(std::make_obj_using_allocator(al, args...))`. In the case of `c(args...)`, the passed arguments are fully specified. But the mapping itself is a bit unclear.